lib/solr/request/first_shard_leader_node_selection_strategy.rb

```diff
@@ -25,7 +19,7 @@ def first_shard_leader_replica_node_for(collection:)
       end
 
       def solr_cloud_active_nodes_urls
```

Comment:
shouldn't leader be excluded from this list of replicas?

Reply:
It should be but it's done with `.flatten.uniq` I didn't change much this strategy, it's refactoring